src/main/java/co/edu/eam/disenosoftware/mitienda/services/OrderProductService.java

```diff
@@ -44,4 +49,74 @@ public void checkOrderProductById(Long idOrderProduct) {
     }
 
   }
+  /**
+   * Repository for find order
+   */
+  @Autowired
+  private OrderRepository ordenRepository;
+
+  /**
+   * Repository for find productStore
+   */
+  @Autowired
+  private ProductStoreRepository productStoreRepository;
+
+  /**
+   * Adding a product to Order
+   *
+   * @param idProduct , Foreign key
+   * @param idOrder , Foreign key
+   * @param quantity , Integer
+   * @return a order or null if not exists
+   */
+  public OrderProduct addingProductToOrderProduct(Long idProduct, Long idOrder, Integer quantity) {
+    final double porcent = 0.1;
+    Order orderToFind = ordenRepository.find(idOrder);
+
+
+    if (orderToFind.getState().equals("canceled") || orderToFind.getState().equals("finished")) {
+
+      throw new BusinessException("El product no puede ser agregado",
+               ErrorCodesEnum.PRODUCT_CAN_NOT_BE_ADDED);
+    }
+
+    Store store = orderToFind.getStore();
+    ProductStore productStoreToFind = productStoreRepository.
+             getProductStoreByIdStoreAndProductId(store.getId(), idProduct);
+
+    if (productStoreToFind == null) {
+
+      throw new BusinessException("La Orden no fue encontrada", ErrorCodesEnum.NOT_ASSOCIATED_STORE);
+    }
+
+    for (int i = 0; i < orderToFind.getProduct().size(); i++) {
```

Comment:
use mejor un foreach
for (int i = 0; i < orderToFind.getProduct().size(); i++) => for(OrderProduct product: orderToFind.getProduct())

Reply:
Listo Profe :D